src/EventListener/ParamConverterListener.php

```diff
@@ -104,6 +104,28 @@ private function autoConfigure(\ReflectionFunctionAbstract $r, Request $request,
         return $configurations;
     }
 
+    private function getParamClassByType(?\ReflectionType $type): ?string
+    {
+        if (\PHP_VERSION_ID < 80000) {
+            return (null === $type || $type->isBuiltin() || !$type instanceof \ReflectionNamedType) ?
+                null : $type->getName();
+        }
+
+        foreach ($type instanceof \ReflectionUnionType ? $type->getTypes() : [$type] as $type) {
+            if (null === $type || $type->isBuiltin() || !$type instanceof \ReflectionNamedType) {
+                continue;
+            }
+
+            $class = new \ReflectionClass($type->getName());
+
+            if (!$class->isInterface() && !$class->isAbstract()) {
+                return $type->getName();
+            }
```

Comment:
Why do you exclude interfaces and abstract classes? This was not the case before your patch. In fact, this change breaks quite a few applications I'm maintaining at the moment.

Reply:
If in the following code we change the order of enumeration of types from Account|EntityInterface to EntityInterface|Account, then without this addition we get an error: Cannot resolve argument $account of "AccountController::read()": Cannot determine controller argument for "AccountController::read()": the $ account argument is type-hinted with the non-existent class or interface:" EntityInterface|Account ".
```
/**
 * @Rest\Route("accounts")
 */
class AccountController extends AbstractFOSRestController implements ApiControllerInterface
{
    /**
     * @Rest\Get("/{id}", requirements={"id"="\d+"})
     *
     * @param Account|EntityInterface $account
     *
     * @return View
     */
    public function read(Account|EntityInterface $account): View
    {
        return $this->accountResponseManager->create($account);
    }
}
```
This addition allows you not to depend on the order of enumerated types, especially since an abstract class and interface cannot be an entity, and a controller action must take an entity

Reply:
> especially since an abstract class and interface cannot be an entity

A Doctrine entity is not the only use-case for a param converter. An application might very well define its own custom param converter that is able to guess a parameter from an interface. This has worked before and we must not break that, especially not with a bugfix release.

I agree that the current implementation does not provide a good algorithm to resolve ambiguous union types. What we've achieved is that `ParamConverterListener` does not error if it comes across a union type. That's great! If you need a better resolution of union types, a PR is more than welcome. But you cannot just throw out support for interfaces just because your app doesn't use them.

Reply:
Excuse for troubling. Thanks for clarifying

Reply:
No worries. 🙂 Do you plan to work on improving the union type support?

Reply:
Yes, of course